nebula/ui/components/MZSignOut.qml

```diff
@@ -9,7 +9,12 @@ MZFooterLink {
     labelText: MZI18n.GlobalSignOut
     fontName: MZTheme.theme.fontBoldFamily
     linkColor: MZTheme.theme.redLinkButton
+
+    property var preLogoutCallback: () => {}
+
     onClicked: () => {
-                   VPNController.logout();
-               }
+        preLogoutCallback();
```

Comment:
what purpose does `preLogoutCallback();` serve here, is it some sort of a delay?

Reply:
It's used to guarantee a certain block of code is run _before_ logout is called.

On CI tests were failing because the metric recording I implemented was not being called before logout destructed the view where the onClick handler was defined. See also: https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8139/commits/769e0481e4212401913e126c3ea77e32c26b8700